### ISSUE TITLE
Files are not visible on browser but are accessible through clicking links

### ISSUE THREAD

**Issue by kalletabur** [owncloud/core#15856]
### Steps to reproduce
Don't know how to reproduce. This appeared already in 7.0.4 version. Now we upgraded to 8.0.2 but the problem is still there

### Expected behaviour
Browsing ownCloud files over web interface should show all the files.

### Actual behaviour
Lot of files are not visible when browsing one web folder. At the beginning of the folder there are lots of folders and files visible, then there is lot of empty space and in the end there is like one file visible again. When hovering over empty space links are shown in URL bar - if you click on link of this file the file can be downloaded.

![image](https://cloud.githubusercontent.com/assets/8167593/7317072/2e6a138a-ea85-11e4-8500-39f0bedc39d4.png)

### Server configuration
**Operating system**:
Debian 7.8

**Web server:**
NGINX (1.2.1-2.2)

**Database:**
MySQL (5.5.43-0)

**PHP version:**
PHP (5.4.4-14)

**ownCloud version:**
ownCloud 8.0.2 (stable) 

**Updated from an older ownCloud or fresh install:**
Updated

**List of activated apps:**
Activity, Deleted files, External storage support, First Run Wizard, LDAP user and group backend, Mail Template Editor, PDF Viewer, Pictures, Server-side Encryption, Share Files, Text Editor, Versions, Video Viewer

**The content of config/config.php:**

```
$CONFIG = array (
  'instanceid' => 'instanceid',
  'passwordsalt' => '',
  'trusted_domains' => 
  array (
    0 => '1.1.1.5',
    1 => 'oc',
  ),
  'datadirectory' => '/var/ownclouddata/',
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => 'dbname',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'prefix',
  'dbuser' => 'dbuser',
  'dbpassword' => '',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'forcessl' => true,
  'loglevel' => '0',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'smtphost',
  'mail_smtpport' => '25',
  'logtimezone' => 'Europe',
  'theme' => '',
  'share_folder' => '/Shared',
  'maintenance' => false,
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'secret' => '',
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
owncloud

**Are you using encryption:** yes/no
yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
AD

#### LDAP configuration (delete this part if not used)

```
|appid|configkey|configvalue|
|user_ldap|cleanUpJobOffset|600|
|user_ldap|enabled|yes|
|user_ldap|has_memberof_filter_support|1|
|user_ldap|home_folder_naming_rule||
|user_ldap|installed_version|0.5.0|
|user_ldap|last_jpegPhoto_lookup|0|
|user_ldap|ldap_agent_password||
|user_ldap|ldap_attributes_for_group_search||
|user_ldap|ldap_attributes_for_user_search||
|user_ldap|ldap_backup_host||
|user_ldap|ldap_backup_port||
|user_ldap|ldap_base|DC=example,DC=local|
|user_ldap|ldap_base_groups|DC=example,DC=local|
|user_ldap|ldap_base_users|DC=example,DC=local|
|user_ldap|ldap_cache_ttl|600|
|user_ldap|ldap_configuration_active|1|
|user_ldap|ldap_display_name|displayname|
|user_ldap|ldap_dn|CN=Comp-Ownc,OU=COM,OU=Company,DC=example,DC=local|
|user_ldap|ldap_email_attr|mail|
|user_ldap|ldap_experienced_admin|0|
|user_ldap|ldap_expert_username_attr|sAMAccountName|
|user_ldap|ldap_expert_uuid_group_attr||
|user_ldap|ldap_expert_uuid_user_attr||
|user_ldap|ldap_group_display_name|cn|
|user_ldap|ldap_group_filter|(&(|(objectclass=group))(|(cn=Dev)(CN=Comp-Ownc)(cn=Comp-admis)))|
|user_ldap|ldap_group_filter_mode|0|
|user_ldap|ldap_group_member_assoc_attribute|member|
|user_ldap|ldap_groupfilter_groups|Comp-DevComp-Owncloud Comp-admis|
|user_ldap|ldap_groupfilter_objectclass|group|
|user_ldap|ldap_host|dc300|
|user_ldap|ldap_login_filter|(&(&(|(objectclass=person))(|(memberof=CN=Comp-Ownc,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-Dev,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-admis,OU=DL,OU=Company,DC=example,DC=local)))(samaccountname=%uid))|
|user_ldap|ldap_login_filter_mode|0|
|user_ldap|ldap_loginfilter_attributes||
|user_ldap|ldap_loginfilter_email|0|
|user_ldap|ldap_loginfilter_username|1|
|user_ldap|ldap_nested_groups|0|
|user_ldap|ldap_nocase|1|
|user_ldap|ldap_override_main_server|0|
|user_ldap|ldap_paging_size|500|
|user_ldap|ldap_port|389|
|user_ldap|ldap_quota_attr||
|user_ldap|ldap_quota_def||
|user_ldap|ldap_tls|0|
|user_ldap|ldap_turn_off_cert_check|1|
|user_ldap|ldap_user_filter_mode|1|
|user_ldap|ldap_userfilter_groups||
|user_ldap|ldap_userfilter_objectclass||
|user_ldap|ldap_userlist_filter|(&(|(objectclass=person))(|(memberof=CN=Comp-Ownc,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-Dev,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-SystemAdministrators,OU=DL,OU=Company,DC=example,DC=local)))|
|user_ldap|ocsid|166061|
|user_ldap|types|authentication|

```

### Client configuration
**Browser:**
Any (chrome, ff, opera)

**Operating system:**
OpenSUSE 13.2 64-bit

#### Web server error log
Adding log entries a about this directory

```
access.log
1.1.1.2 - - [24/Apr/2015:12:12:49 +0300] "GET /index.php/apps/files/ajax/list.php?dir=%2FDocuments%2Fsecurity&sort=name&sortdirection=asc HTTP/1.1" 200 10471 "https://URL2ownCloud.com/index.php/apps/files/?dir=%2FDocuments%2Fsecurity" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36"
1.1.1.2 - - [24/Apr/2015:12:12:49 +0300] "GET /index.php/apps/files/ajax/getstoragestats.php?dir=%2FDocuments%2Fsecurity HTTP/1.1" 200 155 "https://URL2ownCloud.com/index.php/apps/files/?dir=%2FDocuments%2Fsecurity" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36"
1.1.1.2 - - [24/Apr/2015:12:12:59 +0300] "GET /index.php/apps/files/?dir=%2FDocuments%2Fsecurity HTTP/1.1" 200 4976 "https://URL2ownCloud.com/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36"
1.1.1.2 - - [24/Apr/2015:12:13:56 +0300] "GET /core/vendor/jquery/dist/jquery-migrate.min.map HTTP/1.1" 404 47 "https://URL2ownCloud.com/index.php/apps/files/?dir=%2FDocuments%2Fsecurity" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36"
1.1.1.2 - - [24/Apr/2015:12:15:42 +0300] "GET /index.php/apps/files/ajax/download.php?dir=%2FDocuments%2Fsecurity&files=raman.xls HTTP/1.1" 200 72704 "https://URL2ownCloud.com/index.php/apps/files/?dir=%2FDocuments%2Fsecurity" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36"
error.log
2015/04/24 12:13:56 [error] 7760#0: *34187 FastCGI sent in stderr: "Primary script unknown" while reading response header from upstream, client: 1.1.1.2, server: URL2ownCloud.com, request: "GET /core/vendor/jquery/dist/jquery-migrate.min.map HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "URL2ownCloud.com", referrer: "https://URL2ownCloud.com/index.php/apps/files/?dir=%2FDocuments%2Fsecurity"

```

#### ownCloud log (data/owncloud.log)
I'm adding log entries that have related to one file that is not visible but through link the file is accessible
```
{"reqId":"552b73a87fc56","app":"files_encryption","message":"delShareKey: delete share key: \/user_owner\/files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_1.shareKey","level":0,"time":"2015-04-13T10:43:36+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73ab1a1a5","app":"files_encryption","message":"delShareKey: delete share key: \/user_owner\/files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_2.shareKey","level":0,"time":"2015-04-13T10:43:39+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73b4d5eb5","app":"OC\\Files\\Cache\\Scanner","message":"!!! No reuse of etag for 'files_encryption\/keyfiles\/Documents\/security\/pp_v02-1.docx.key' !!! \ncache: Array\n(\n    [fileid] => 48812\n    [storage] => home::user_owner\n    [path] => files_encryption\/keyfiles\/Documents\/security\/pp_v02-1.docx.key\n    [parent] => 48596\n    [name] => pp_v02-1.docx.key\n    [mimetype] => application\/x-iwork-keynote-sffkey\n    [mimepart] => application\n    [size] => 244\n    [mtime] => 1411549988\n    [storage_mtime] => 1411549988\n    [encrypted] => \n    [unencrypted_size] => 0\n    [etag] => 544ed3d1d30a3\n    [permissions] => 27\n)\n \ndata: Array\n(\n    [mimetype] => application\/x-iwork-keynote-sffkey\n    [mtime] => 1428911030\n    [size] => 244\n    [etag] => 552b73b65be27\n    [storage_mtime] => 1428911030\n    [permissions] => 27\n    [parent] => 48596\n)\n","level":0,"time":"2015-04-13T10:43:50+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73b4d5eb5","app":"OC\\Files\\Cache\\Scanner","message":"!!! No reuse of etag for 'files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_owner.shareKey' !!! \ncache: Array\n(\n    [fileid] => 49575\n    [storage] => home::user_owner\n    [path] => files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_owner.shareKey\n    [parent] => 49204\n    [name] => pp_v02-1.docx.user_owner.shareKey\n    [mimetype] => application\/octet-stream\n    [mimepart] => application\n    [size] => 512\n    [mtime] => 1411549988\n    [storage_mtime] => 1411549988\n    [encrypted] => \n    [unencrypted_size] => 0\n    [etag] => 544ed3d63bb57\n    [permissions] => 27\n)\n \ndata: Array\n(\n    [mimetype] => application\/octet-stream\n    [mtime] => 1428911030\n    [size] => 512\n    [etag] => 552b73b6621b0\n    [storage_mtime] => 1428911030\n    [permissions] => 27\n    [parent] => 49204\n)\n","level":0,"time":"2015-04-13T10:43:50+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73bd4a570","app":"OC\\Files\\Cache\\Scanner","message":"!!! No reuse of etag for 'files_encryption\/keyfiles\/Documents\/security\/pp_v02-1.docx.key' !!! \ncache: Array\n(\n    [fileid] => 48812\n    [storage] => home::user_owner\n    [path] => files_encryption\/keyfiles\/Documents\/security\/pp_v02-1.docx.key\n    [parent] => 48596\n    [name] => pp_v02-1.docx.key\n    [mimetype] => application\/x-iwork-keynote-sffkey\n    [mimepart] => application\n    [size] => 244\n    [mtime] => 1428911030\n    [storage_mtime] => 1428911030\n    [encrypted] => \n    [unencrypted_size] => 0\n    [etag] => 552b73b65be27\n    [permissions] => 27\n)\n \ndata: Array\n(\n    [mimetype] => application\/x-iwork-keynote-sffkey\n    [mtime] => 1428911038\n    [size] => 244\n    [etag] => 552b73bed5a58\n    [storage_mtime] => 1428911038\n    [permissions] => 27\n    [parent] => 48596\n)\n","level":0,"time":"2015-04-13T10:43:58+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73bd4a570","app":"OC\\Files\\Cache\\Scanner","message":"!!! No reuse of etag for 'files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_owner.shareKey' !!! \ncache: Array\n(\n    [fileid] => 49575\n    [storage] => home::user_owner\n    [path] => files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_owner.shareKey\n    [parent] => 49204\n    [name] => pp_v02-1.docx.user_owner.shareKey\n    [mimetype] => application\/octet-stream\n    [mimepart] => application\n    [size] => 512\n    [mtime] => 1428911030\n    [storage_mtime] => 1428911030\n    [encrypted] => \n    [unencrypted_size] => 0\n    [etag] => 552b73b6621b0\n    [permissions] => 27\n)\n \ndata: Array\n(\n    [mimetype] => application\/octet-stream\n    [mtime] => 1428911038\n    [size] => 512\n    [etag] => 552b73bedd64e\n    [storage_mtime] => 1428911038\n    [permissions] => 27\n    [parent] => 49204\n)\n","level":0,"time":"2015-04-13T10:43:58+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}
{"reqId":"552b73bd4a570","app":"OC\\Files\\Cache\\Scanner","message":"!!! No reuse of etag for 'files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_2.shareKey' !!! \ncache: Array\n(\n    [fileid] => 151499\n    [storage] => home::user_owner\n    [path] => files_encryption\/share-keys\/Documents\/security\/pp_v02-1.docx.user_2.shareKey\n    [parent] => 49204\n    [name] => pp_v02-1.docx.user_2.shareKey\n    [mimetype] => application\/octet-stream\n    [mimepart] => application\n    [size] => 512\n    [mtime] => 1428911030\n    [storage_mtime] => 1428911030\n    [encrypted] => \n    [unencrypted_size] => 0\n    [etag] => 552b73b669c72\n    [permissions] => 27\n)\n \ndata: Array\n(\n    [mimetype] => application\/octet-stream\n    [mtime] => 1428911038\n    [size] => 512\n    [etag] => 552b73bee3420\n    [storage_mtime] => 1428911038\n    [permissions] => 27\n    [parent] => 49204\n)\n","level":0,"time":"2015-04-13T10:43:58+03:00","method":"POST","url":"\/index.php\/core\/ajax\/share.php"}

```

**Comment by eagles051387** [comment]
I can confirm I have the same issue but my install is on ubuntu 14.04 with the package of OC8.0 at the time from the open suse build system. Mine was not an upgrade but a clean installation.

**Comment by PVince81** [comment]
Did you guys disable "Share API" in the admin page ? If you wanted to disable sharing, then please also deactivate the "Sharing" app.
Do you see any Javascript errors when trying to scroll down in that folder ?

**Comment by kalletabur** [comment]
I haven't disabled it. Sharing API is enabled and we use sharing.
No javascript errors while scrolling.

**Comment by PVince81** [comment]
@kalletabur can you use the browser's DOM inspector on the white space to see if the elements exist there ? It could be some kind of weird rendering problem

**Comment by atroxix** [comment]
When looking at Elements under Chrome's developer tools it can be seen tha at some point all lines forward are created with some transparent class:
```html
<tr data-id="145572" data-type="file" data-size="25600" data-file="..." data-mime="application/msword" data-mtime="..." data-etag="..." data-permissions="27" data-share-permissions="19" class="appear transparent">
```

I also see 2 errors:
* Uncaught TypeError: Cannot read property 'toString' of undefined
* jquery-migrate.min.map:1 GET /core/vendor/jquery/dist/jquery-migrate.min.map 404 (Not Found)

**Comment by PVince81** [comment]
The transparent class is set on the rows to create an animation by removing it.
But something is preventing the JS code to do its work.

@atroxix you too confirm that you didn't disable "Share API" in the admin page ? (because that's a known issue: https://github.com/owncloud/core/issues/15186)

Are you able to get the stack trace of that "toString" error ? (maybe with the debugger)

**Comment by atroxix** [comment]
BTW, Kalle was reporting my error, so I just took over debugging my system myself ;)

So yes, haven't disabled "Share API".

About "toString" error - you mean this?

```
Uncaught TypeError: Cannot read property 'toString' of undefined
escapeHTML @ js.js?v=66bd8398957a816353d73de6cde9dd14:47
(anonymous function) @ share.js?v=66bd8398957a816353d73de6cde9dd14:164
(anonymous function) @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.extend.each @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:4
x.fn.x.each @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:4
x.fn.extend.domManip @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.fn.extend.append @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5O
CA.Sharing.Util.updateFileActionIcon @ share.js?v=66bd8398957a816353d73de6cde9dd14:163
(anonymous function) @ share.js?v=66bd8398957a816353d73de6cde9dd14:70
_.each._.forEach @ underscore.js?v=66bd8398957a816353d73de6cde9dd14:79
updateIcons @ share.js?v=66bd8398957a816353d73de6cde9dd14:69
(anonymous function) @ share.js?v=66bd8398957a816353d73de6cde9dd14:84
x.event.dispatch @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.event.add.v.handle @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.event.trigger @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
e.event.trigger @ jquery-migrate.min.js?v=66bd8398957a816353d73de6cde9dd14:2
(anonymous function) @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.extend.each @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:4
x.fn.x.each @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:4
x.fn.extend.trigger @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
FileList._nextPage @ filelist.js?v=66bd8398957a816353d73de6cde9dd14:600
FileList._onScroll @ filelist.js?v=66bd8398957a816353d73de6cde9dd14:482
x.event.dispatch @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
x.event.add.v.handle @ jquery.min.js?v=66bd8398957a816353d73de6cde9dd14:5
```

**Comment by PVince81** [comment]
@atroxix ok so the error happens here: https://github.com/owncloud/core/blob/v8.0.2/apps/files_sharing/js/share.js#L164

I guess maybe the attribute "data-share-owner" is missing on some rows in the list.
But it might not be the core issue, because if other JS are interrupted due to an error, they wouldn't get a chance to add that attribute.

Does this stack appear when reloading the file list page (complete reload of the web page) ?

Also, do you see any failing (red) calls in the network console ?

**Comment by atroxix** [comment]
The error does appear when I start scrolling (not even hovering over any of this empty area).

Under Network I see 1 red line:
* https://.../core/vendor/jquery/dist/jquery-migrate.min.map: File not found.

**Comment by PVince81** [comment]
Would you guys mind posting the content of your "oc_appconfig" table ? Make sure to censor private data, if applicable. It might help to find out config variations which might help reproduce the issue.

**Comment by atroxix** [comment]
Here's the info from SQL dump (I hope I managed to get rid of all sensitive values there):

```
CREATE TABLE `oc_appconfig` (
  `appid` varchar(32) COLLATE utf8_bin NOT NULL DEFAULT '',
  `configkey` varchar(64) COLLATE utf8_bin NOT NULL DEFAULT '',
  `configvalue` longtext COLLATE utf8_bin,
  PRIMARY KEY (`appid`,`configkey`),
  KEY `appconfig_config_key_index` (`configkey`),
  KEY `appconfig_appid_key` (`appid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `oc_appconfig`
--

LOCK TABLES `oc_appconfig` WRITE;
/*!40000 ALTER TABLE `oc_appconfig` DISABLE KEYS */;
INSERT INTO `oc_appconfig` VALUES
('activity','enabled','yes'),
('activity','installed_version','1.2.0'),
('activity','ocsid','166038'),
('activity','types','filesystem'),
('admin_dependencies_chk','enabled','no'),
('admin_dependencies_chk','installed_version','0.01'),
('admin_dependencies_chk','types',''),
('backgroundjob','lastjob','3'),
('calendar','enabled','no'),
('calendar','installed_version','0.6.3'),
('calendar','types',''),
('contacts','enabled','no'),
('contacts','installed_version','0.3'),
('contacts','types',''),
('core','global_cache_gc_lastrun','1430395414'),
('core','installedat','1404397732.3113'),
('core','lastcron','1430396161'),
('core','lastupdateResult','{\"version\":{},\"versionstring\":{},\"url\":{},\"web\":{}}'),
('core','lastupdatedat','1430294259'),
('core','public_caldav','calendar/share.php'),
('core','public_calendar','calendar/share.php'),
('core','public_documents','documents/public.php'),
('core','public_files','files_sharing/public.php'),
('core','public_gallery','gallery/public.php'),
('core','public_webdav','files_sharing/publicwebdav.php'),
('core','remote_caldav','calendar/appinfo/remote.php'),
('core','remote_calendar','calendar/appinfo/remote.php'),
('core','remote_carddav','contacts/appinfo/remote.php'),
('core','remote_contacts','contacts/appinfo/remote.php'),
('core','remote_core.css','/core/minimizer.php'),
('core','remote_core.js','/core/minimizer.php'),
('core','remote_files','files/appinfo/remote.php'),
('core','remote_filesync','files/appinfo/filesync.php'),
('core','remote_webdav','files/appinfo/remote.php'),
('core','repairlegacystoragesdone','yes'),
('core','shareapi_allow_links','yes'),
('core','shareapi_allow_mail_notification','yes'),
('core','shareapi_allow_public_notification','yes'),
('core','shareapi_allow_public_upload','yes'),
('core','shareapi_allow_resharing','no'),
('core','shareapi_default_expire_date','yes'),
('core','shareapi_enforce_links_password','yes'),
('core','shareapi_expire_after_n_days','90'),
('documents','enabled','no'),
('documents','installed_version','0.8.2'),
('documents','ocsid','166045'),
('documents','types',''),
('files','backgroundwatcher_previous_file','736'),
('files','backgroundwatcher_previous_folder','2533'),
('files','default_quota','1 GB'),
('files','enabled','yes'),
('files','installed_version','1.1.9'),
('files','types','filesystem'),
('files_encryption','enabled','yes'),
('files_encryption','installed_version','0.7.1'),
('files_encryption','ocsid','166047'),
('files_encryption','publicShareKeyId','pubShare_cddf305d'),
('files_encryption','recoveryAdminEnabled','1'),
('files_encryption','recoveryKeyId','recovery_83cfb34c'),
('files_encryption','types','filesystem'),
('files_pdfviewer','enabled','yes'),
('files_pdfviewer','installed_version','0.7'),
('files_pdfviewer','ocsid','166049'),
('files_pdfviewer','types',''),
('files_sharing','enabled','yes'),
('files_sharing','installed_version','0.6.1'),
('files_sharing','ocsid','166050'),
('files_sharing','types','filesystem'),
('files_texteditor','enabled','yes'),
('files_texteditor','installed_version','0.4'),
('files_texteditor','ocsid','166051'),
('files_texteditor','types',''),
('files_trashbin','enabled','yes'),
('files_trashbin','installed_version','0.6.2'),
('files_trashbin','ocsid','166052'),
('files_trashbin','types','filesystem'),
('files_versions','enabled','yes'),
('files_versions','installed_version','1.0.5'),
('files_versions','ocsid','166053'),
('files_versions','types','filesystem'),
('files_videoviewer','enabled','yes'),
('files_videoviewer','installed_version','0.1.3'),
('files_videoviewer','ocsid','166054'),
('files_videoviewer','types',''),
('firstrunwizard','enabled','yes'),
('firstrunwizard','installed_version','1.1'),
('firstrunwizard','ocsid','166055'),
('firstrunwizard','types',''),
('gallery','enabled','yes'),
('gallery','installed_version','0.6.0'),
('gallery','ocsid','166056'),
('gallery','types',''),
('search_lucene','enabled','no'),
('search_lucene','installed_version','0.5.3'),
('search_lucene','ocsid','166057'),
('search_lucene','types','filesystem'),
('updater','enabled','no'),
('updater','installed_version','0.4'),
('updater','ocsid','166059'),
('updater','types',''),
('user_ldap','cleanUpJobOffset','50'),
('user_ldap','enabled','yes'),
('user_ldap','installed_version','0.5.0'),
('user_ldap','ocsid','166061'),
('user_ldap','s01has_memberof_filter_support','1'),
('user_ldap','s01home_folder_naming_rule',''),
('user_ldap','s01last_jpegPhoto_lookup','0'),
('user_ldap','s01ldap_agent_password','...'),
('user_ldap','s01ldap_attributes_for_group_search',''),
('user_ldap','s01ldap_attributes_for_user_search',''),
('user_ldap','s01ldap_backup_host',''),
('user_ldap','s01ldap_backup_port',''),
('user_ldap','s01ldap_base','...'),
('user_ldap','s01ldap_base_groups','...'),
('user_ldap','s01ldap_base_users','...'),
('user_ldap','s01ldap_cache_ttl','600'),
('user_ldap','s01ldap_configuration_active','1'),
('user_ldap','s01ldap_display_name','displayname'),
('user_ldap','s01ldap_dn','...'),
('user_ldap','s01ldap_email_attr','mail'),
('user_ldap','s01ldap_experienced_admin','0'),
('user_ldap','s01ldap_expert_username_attr','sAMAccountName'),
('user_ldap','s01ldap_expert_uuid_group_attr',''),
('user_ldap','s01ldap_expert_uuid_user_attr',''),
('user_ldap','s01ldap_group_display_name','cn'),
('user_ldap','s01ldap_group_filter','...'),
('user_ldap','s01ldap_group_filter_mode','0'),
('user_ldap','s01ldap_group_member_assoc_attribute','member'),
('user_ldap','s01ldap_groupfilter_groups','...'),
('user_ldap','s01ldap_groupfilter_objectclass','group'),
('user_ldap','s01ldap_host','...'),
('user_ldap','s01ldap_login_filter','...'),
('user_ldap','s01ldap_login_filter_mode','1'),
('user_ldap','s01ldap_loginfilter_attributes',''),
('user_ldap','s01ldap_loginfilter_email','0'),
('user_ldap','s01ldap_loginfilter_username','1'),
('user_ldap','s01ldap_nested_groups','0'),
('user_ldap','s01ldap_nocase','1'),
('user_ldap','s01ldap_override_main_server','0'),
('user_ldap','s01ldap_paging_size','500'),
('user_ldap','s01ldap_port','389'),
('user_ldap','s01ldap_quota_attr',''),
('user_ldap','s01ldap_quota_def',''),
('user_ldap','s01ldap_tls','0'),
('user_ldap','s01ldap_turn_off_cert_check','1'),
('user_ldap','s01ldap_user_filter_mode','1'),
('user_ldap','s01ldap_userfilter_groups',''),
('user_ldap','s01ldap_userfilter_objectclass','person'),
('user_ldap','s01ldap_userlist_filter','...'),
('user_ldap','types','authentication');
/*!40000 ALTER TABLE `oc_appconfig` ENABLE KEYS */;
UNLOCK TABLES;
```

**Comment by PVince81** [comment]
Okay, the things you guys have in common so far:
- LDAP enabled
- encryption enabled

@atroxix I see you enforced expire date and link password (which is a nice thing :wink:) and disabled reshares. Let me try this locally.

**Comment by PVince81** [comment]
Just noticed that @kalletabur has nginx. @atroxix how about you ? (just to find more common stuff)

**Comment by atroxix** [comment]
Kalle first created the issue on behalf of me, so we are speaking about the same issue ;)

**Comment by PVince81** [comment]
Ah! Good to know :smile:

**Comment by PVince81** [comment]
Still out of luck at the moment. I tried to replicate your sharing settings and also use LDAP users, but I cannot reproduce the issue. I tried with a long list of files where shared folders/files appear at the bottom of the list. Scrolling still works.

Does this only happen for a specific user ?
If yes, what is special about this user ? Does the folder contain any outgoing or incoming shares ?

**Comment by PVince81** [comment]
well, I also have encryption and the other apps enabled too, with the recovery key.

**Comment by PVince81** [comment]
@eagles051387 it would be helpful if you could also share more details. It could help to cross-reference the config with the one from @atroxix / @kalletabur to see what they have in common. Do you also have LDAP ?

**Comment by atroxix** [comment]
What's special about my user:
* I've got tons of files/folders shared with very many people; there are also shared files within shared folders etc.
* I can't imagine how can a folder contain some other data shared with myself, so I guess there are no "incoming shares" :)
* Although we have encryption and recovery key enabled, I haven't enabled recovery key for my user.

**Comment by eagles051387** [comment]
The difference then in my setup is im not using sharing as its my own
personal OC server plus i have encryption enabled also I am using apache as
my webserver

On Thu, Apr 30, 2015 at 8:51 PM, Silver <notifications@github.com> wrote:

> What's special about my user:
>
>    - I've got tons of files/folders shared with very many people; there
>    are also shared files within shared folders etc.
>    - I can't imagine how can a folder contain some other data shared with
>    myself, so I guess there are no "incoming shares" :)
>    - Although we have encryption and recovery key enabled, I haven't
>    enabled recovery key for my user.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/owncloud/core/issues/15856#issuecomment-97923600>.
>



-- 
Jonathan Aquilina

**Comment by atroxix** [comment]
We discovered that when I change ordering to either by size or date then the files are shown. When I change it back to Name then the files are hidden again.

**Comment by PVince81** [comment]
That is really weird...
When you sort by size or date, did you scroll all the way to the bottom to make sure that ALL files were shown properly ? And then when you click on the "Name" column some files disappear again ? (not a full page refresh)

**Comment by PVince81** [comment]
Does it also happen if you share a folder with link and open the link in the browser ?
If yes you could create a test folder and post the link here (or send it to me by email to pvince81 at owncloud dot com) so I could have a look to see if I can find more clues.

**Comment by atroxix** [comment]
* Yes, all the files are shown when changing ordering
* Yes, the files disappear once I order by Name again. But they are shown when ordered descending
* Yes, behavior is the same when shared by link
* No, I cannot give you access to any test folder on that server as it is not publicly accessible (by network)

**Comment by PVince81** [comment]
Okay, I understand.

How about you @eagles051387 ?

**Comment by phaenomenon** [comment]
I am facing the same issue here, running Owncloud 8.0.4-0 on Debian Jessie 64bit. I have a CIFS/SMB external storage attached and while browsing onto it, I see only the first 17 folders in the browser, although the current directory has much more folders in there. I tested with different workstations (Windows + Linux) and following browsers: Internet Explorer, Mozilla Firefox, Chromium, Iceweasel.

I did find out following. While in Firefox, I press F12 for debugging the current web page. If you look at the following screenshot
![clipboard01](https://cloud.githubusercontent.com/assets/12893756/8475885/f000dfe4-20bb-11e5-9445-305abae965fd.jpg)

you will realize that the working folders (in my example called "Großbritanninen, Indonesien, Irland" don't have the class "appear-transparent" assigned. The next folder name "Island" and the following folders are INVISIBLE because they have the class "appear-transparent" assigned which is using the CSS value "opacity: 0;" within apps.css line 515.

As soon as I am manually setting the opacity: to value '1' the invisible folders become visible and they should be. IMHO the solution is just simple ==> Remove the class "appear-transparent" from these folders. Why is this class assigned to the 17th folder and to the following ones?

Please anyone check and fix that.

Thank you
Phaenomenon

**Comment by PVince81** [comment]
@phaenomenon would it be possible for you to create such a folder with dummy files and share it with public link so I could have a look ?

I cannot reproduce this locally and there isn't enough information yet to be able to fix it.

**Comment by PVince81** [comment]
The only explanation so far would be that some error is happening in the middle of a JS process which prevents it to finish making the new files visible. But if the JS console doesn't contain anything, it will be difficult to guess what was wrong.

**Comment by PVince81** [comment]
Note: the class "appear-transparent" is there to create the fade-in effect of files when you scroll down or create new files. The class should normally be removed here:
https://github.com/owncloud/core/blob/v8.0.4/apps/files/js/filelist.js#L874

To create this CSS3 effect you need to first set the class then remove it. As you can see the class is being set here https://github.com/owncloud/core/blob/v8.0.4/apps/files/js/filelist.js#L591 and removed later here https://github.com/owncloud/core/blob/v8.0.4/apps/files/js/filelist.js#L607.

Can you add `console.log('set appear'`);` and `console.log('remove appear')`) in both locations ? Goal is to find out whether that code is executed.
You will need to add `define('DEBUG', 1)` to your config PHP to see the log messages. (don't forget to remove it afterwards)

The only thing that appears before the animation is this: https://github.com/owncloud/core/blob/v8.0.4/apps/files/js/filelist.js#L600. This is an event that will let third party apps register file actions.
Did you enable any third party app that provides file actions / file viewers ? (ex: PDF viewer, etc).
Maybe such app is having an error.

**Comment by PVince81** [comment]
Ok or simpler: disable all third party apps related to files (file viewer, etc)

I'll try and provide a patch to verify my hypothesis.

**Comment by phaenomenon** [comment]
Hi PVince81, our policy unfortunately does not allow the sharing and I cannot create a public share which you could access on our Owncloud server. But I just tried following: while I am logged in at Owncloud through Web, within my external_storage called "foo" I create manually folder names and call them "folder1", "folder2", "folder3", etc....

The number of folder (17) I told you in my last comment is not a static thing. In my actual test after the 11th folder the following folders were invisible. Then I did the same test again but before starting creating test folder, I pulled the splitting window line (because I am still in F12 debuggin mode within Firefox) to the top, so the browser content window will get smaller as you see here...
![clipboard02](https://cloud.githubusercontent.com/assets/12893756/8476118/5ac57c98-20be-11e5-9f1c-5b69c529f653.jpg)

This time after folder6 the issue occured. In my conclusion that means: All folders that were not visible in the initial view area in browser window, were applied with "appear transparent" class. 

HTH

**Comment by PVince81** [comment]
@phaenomenon I managed to reproduce it locally by making "fileActionsReady" throw an exception.

So either you have a third party app that listens to this even, or there is a problem with sharing, see https://github.com/owncloud/core/blob/v8.0.4/apps/files_sharing/js/share.js#L61

Are you able to debug the JS into https://github.com/owncloud/core/blob/v8.0.4/apps/files_sharing/js/share.js#L61 and see how far it goes ?

I suspect something is wrong with shares and causes that function to fail in the middle.

**Comment by PVince81** [comment]
Ok, I'm preparing a patch to make such failure more graceful. You'd be able to continue using the file list with it. (will post it soon)

However we still need to investigate the share issue which is the root cause.

**Comment by PVince81** [comment]
This PR should make the list usable again https://github.com/owncloud/core/issues/17341.
Please note that this does not fix the root cause, it only mitigates it with a more graceful failure.
If you have `define('DEBUG', 1)` in config.php and this patch applied, it should show the error in the JS console.

If no one here is able to patch their system yet, we'll need to find another way to analyze the root cause. It is likely to be related to broken shares.

**Comment by PVince81** [comment]
Hah, I think that stack trace might contain a clue: https://github.com/owncloud/core/issues/15856#issuecomment-97735879

**Comment by PVince81** [comment]
Okay, we're getting closer. The screenshot from @phaenomenon shows that the "data-share-owner" attribute is missing on the file rows: https://github.com/owncloud/core/issues/15856#issuecomment-118004905

I assume that you are the recipient of this shared folder ?

**Comment by PVince81** [comment]
This stack https://github.com/owncloud/core/issues/15856#issuecomment-97735879 can only happen if the current user is not allowed to share at all. This means that all of you have set the special option "Exclude groups from sharing" in the admin page. Can you confirm ?

I'll see if I can recreate the situation locally.

**Comment by phaenomenon** [comment]
Hey all, it works now!!! Thanks to PVince81 comment about the "shared folder" which pointed me to the right direction. Because I remember I didn't want to have my users SHARE stuff I had disabled that right after I initially installed Owncloud. So I just checked in the "Administration" settings. I had unchecked the option "Allow applications to use the share API" so this was NOT active. The administrator have no clue that such a setting would have such a massive effect in real. After I "checked" this setting (I have ticked only "Allow applications to use the share API" under Administration-->Sharing) it works fine and I see ALL folders. 

It would be nice if that could be fixed, at least, it should not be in relation to this setting because it's confusing.

HTH,

cheers
Phaenomenon

**Comment by PVince81** [comment]
@phaenomenon glad it fixed that for you, so the bug you had is this https://github.com/owncloud/core/issues/15186

Unfortunately the other reporters here do have it enabled, so there might be yet another issue.

**Comment by PVince81** [comment]
I tried using the "Exclude groups from sharing" option but couldn't reproduce the original issue from @kalletabur and @atroxix.

I'll try and dig into the code to find out when the "data-share-owner" attribute is missing.

**Comment by PVince81** [comment]
So far what we know:
- resharing is disabled
- the folder in question is not a shared folder (not incoming), as I don't see any permanent "Share" links visible in the screenshot (also confirmed here https://github.com/owncloud/core/issues/15856#issuecomment-97923600)
- the problem is caused by a missing attribute "data-share-owner" in a code path that is triggered when the user has no share permissions

**Comment by PVince81** [comment]
I see here that the display name of the share owner would be read: https://github.com/owncloud/core/blob/v8.0.4/apps/files_sharing/lib/share/folder.php#L43

Now I see that if the display name would be empty or a value like "0", the attribute would not be set at all.

@atroxix do you have a user called "0" ? Another guess is that LDAP doesn't return / doesn't contain any display name for the users ? It would explain the symptoms.

**Comment by PVince81** [comment]
AHA got it reproduced by creating a share at first, and the disallowing sharing for that group.
Steps will follow.

**Comment by PVince81** [comment]
### Steps to reproduce

1. Login as admin
1. Create two users "user1" and "user2"
1. Add "user1" to group "group1"
1. Login as "user1"
1. Create a folder "test"
1. Upload 20-30 files into "test"
1. Scroll down and share any file on the second page (best is in the middle), share it with "user2"
1. Login as admin
1. Go to the admin page and set "Exclude groups from sharing" to "group1".
1. Login again as "user1"
1. Go into "test" and scroll down to the shared file

### Expected result
List appears after scroll

### Actual result
The list stops right before the file that was shared before the share interdiction.

JS console says:
```
Uncaught TypeError: Cannot read property 'toString' of undefinedescapeHTML @ js.js?v=c0339554766dde1a3b9b19cdb9877b06:47(anonymous function) @ share.js?v=c0339554766dde1a3b9b19cdb9877b06:164(anonymous function) @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.extend.each @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:4x.fn.x.each @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:4x.fn.extend.domManip @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.fn.extend.append @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5OCA.Sharing.Util.updateFileActionIcon @ share.js?v=c0339554766dde1a3b9b19cdb9877b06:163(anonymous function) @ share.js?v=c0339554766dde1a3b9b19cdb9877b06:70_.each._.forEach @ underscore.js?v=c0339554766dde1a3b9b19cdb9877b06:79updateIcons @ share.js?v=c0339554766dde1a3b9b19cdb9877b06:69(anonymous function) @ share.js?v=c0339554766dde1a3b9b19cdb9877b06:84x.event.dispatch @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.event.add.v.handle @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.event.trigger @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5e.event.trigger @ jquery-migrate.min.js?v=c0339554766dde1a3b9b19cdb9877b06:2(anonymous function) @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.extend.each @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:4x.fn.x.each @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:4x.fn.extend.trigger @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5FileList._nextPage @ filelist.js?v=c0339554766dde1a3b9b19cdb9877b06:600FileList._onScroll @ filelist.js?v=c0339554766dde1a3b9b19cdb9877b06:482x.event.dispatch @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5x.event.add.v.handle @ jquery.min.js?v=c0339554766dde1a3b9b19cdb9877b06:5
```

So basically there is a contradiction: the user still has an existing share but is now not allowed to share any more.

**Comment by PVince81** [comment]
Raised https://github.com/owncloud/core/issues/17346 for the issue about stray "forbidden" shares.

I'll provide a quickfix to make sure the JS code isn't confused by such situations.

**Comment by PVince81** [comment]
Quickfix here: https://github.com/owncloud/core/pull/17349

**Comment by atroxix** [comment]
That quickfix indeed fixes my issue at least on our test server with OwnCloud 8.0.2.

**Comment by atroxix** [comment]
It seems the same root cause also causes the sync client in other computers to fail synchronization. Is there any other issue about this?

**Comment by PVince81** [comment]
Not that I'm aware of. The JS fix wouldn't help with this.
Might opening a new ticket and post the owncloud.log matching the problem ?

**Comment by atroxix** [comment]
OK: #17444

**Comment by PVince81** [comment]
@atroxix thanks a lot
